src/app.rb

```diff
@@ -0,0 +1,107 @@
+require_relative './student'
+require_relative './teacher'
+require_relative './book'
+require_relative './rental'
+
+class App
+  def initialize(books = [], people = [], rentals = [])
+    @books = books
+    @people = people
+    @rentals = rentals
+  end
+
+  def list_all_books
+    @books.each_with_index do |_books, index|
+      puts "#{index + 1}) title: #{_books.title} - author: #{_books.author}"
+    end
+  end
+
+  def list_all_people
+    @people.each do |person|
+      puts "[#{person.class}] ID: #{person.id} Name: #{person.name} - Age: #{person.age}"
+    end
+  end
+
+  def create_person
+    flag = true
+    while flag
+      puts 'Do you want to create a student (1) or a teacher (2)? [Input a number]:'
+      choice = gets.chomp.to_i
+      case choice
+      when 1
+        create_student
+        flag = false
+      when 2
+        create_teacher
+        flag = false
+      else
+        puts 'Invalid input. Please try again.'
+      end
+    end
+  end
+
+  def create_student
+    puts 'Age: '
+    age = gets.chomp.to_i
+    puts 'Name: '
+    name = gets.chomp.capitalize
+    puts 'Has parent permission? (y/n): '
+    parent_permission = gets.chomp.downcase == 'y'
+    student = Student.new(age, name, parent_permission: parent_permission)
+    @people << student unless @people.include?(student)
+    puts 'Student created successfully.'
+  end
+
+  def create_teacher
+    puts 'Age: '
+    age = gets.chomp.to_i
+    puts 'Name: '
+    name = gets.chomp.capitalize
+    puts 'specialization: '
+    specialization = gets.chomp.capitalize
+    teacher = Teacher.new(age, specialization, name)
+    @people << teacher unless @people.include?(teacher)
+    puts 'Teacher created successfully.'
+  end
+
+  def create_book
+    print 'Enter the title of the book: '
+    title = gets.chomp.capitalize
+    print 'Enter the author of the book: '
+    author = gets.chomp.capitalize
+    book = Book.new(title, author)
+    @books << book unless @books.include?(book)
+    puts 'Book created successfully.'
+  end
+
+  def create_rental
+    puts 'Select a book from the following list by number:'
+    list_all_books
+    print 'Enter the number of the book: '
+    book_index = gets.chomp.to_i - 1
+    puts 'Select a person from the following list by number: (not id)'
```

Comment:
- [ ] When creating a rental, kindly number the persons so that a user can write the correct number and not guess it.
- Your console:
![Annotation 2022-06-17 061150](https://user-images.githubusercontent.com/85113567/174229845-e43ccc65-5048-4349-ab3f-13996ffccc59.jpg)